pkg/registry/apis/query/query.go

```diff
@@ -105,14 +104,14 @@ func (r *queryREST) Connect(ctx context.Context, name string, opts runtime.Objec
 					Message: "datasource not found",
 				}}
 			}
-			responder.Error(convertToK8sError(err))
+			responder.Error(err)
```

Comment:
🎉 the errors are now returned properly -- rather than getting returned as 500, it is now 400